controlplane/rosa/controllers/rosacontrolplane_controller.go

```diff
@@ -276,6 +276,14 @@ func (r *ROSAControlPlaneReconciler) reconcileNormal(ctx context.Context, rosaSc
 		ExpirationTimestamp(time.Now().Add(1 * time.Hour)).
 		Hypershift(cmv1.NewHypershift().Enabled(true))
 
+	etcdEncryption := rosaScope.ControlPlane.Spec.EtcdEncryption
+	// If EtcdEncryptionKMSArn is set, it will override etcdEncryption flag to true.
```

Comment:
I believe better not to override the etcdEncryption flag, if the etcdEncryption set to false it is false. Otherwise the flag has no sense.

Reply:
The EtcdEncryptionKMSArn  overriding the etcdEncryption flag is following ROSA CLI routine. Should we follow up it

https://github.com/openshift/rosa/blob/eaae3fad9fcf36b9954489de0160c652117f9742/cmd/create/cluster/cmd.go#L457-L462

Reply:
since EtcdEncryption can't be set true without providing a kms key ARN, I don't think we need this bool field at all.
Either the KMSKeyArn field is set -> we set encryption to true internally or its not set -> we set encryption to false

Reply:
The only case I can see is during testing you may don't want to delete the arn kms key and want to not apply kms encryption. Otherwise agree, we don't need the the flag at all.

Reply:
agree. I will remove the etcdEncryption flag from the spec